examples/favorite/src/index.ts

```diff
@@ -0,0 +1,218 @@
+
+      const item = list.items.value.find((item) => item.element === itemElement);
+      if (!item?.url?.pathname) return;
+
+      const favorites = (favoritedItemsStore[instance] ||= []);
+
+      const index = favorites.indexOf(item.url.pathname);
+
+      if (index >= 0) {
+        if (favoriteButtonRemove || favoriteButtonToggle) {
+          favorites.splice(index, 1);
+        }
+      } else {
+        if (favoriteButtonAdd || favoriteButtonToggle) {
+          favorites.push(item.url.pathname);
+        }
+      }
+    },
+    { signal }
+  );
+
+  const listCleanup = isDisplay
+    ? initDisplayerFavoritesList(list, instance)
+    : initControllerFavoritesList(list, instance);
+
+  return () => {
+    listCleanup();
+    controller.abort();
+  };
+};
+
+/**
+ * Initializes a list that controls the favorited items.
+ * @param list
+ * @returns A cleanup function
+ */
+const initControllerFavoritesList = (list: List, instance: string) => {
+  const handler = debounce(
+    ({
+      favoritedItems = favoritedItemsStore[instance] || [],
+      items = list.items.value,
+    }: {
+      favoritedItems?: string[];
+      items?: ListItem[];
+    }) => {
+      for (const item of items) {
+        if (!item.url?.pathname) continue;
+
+        const isFavorited = favoritedItems.includes(item.url.pathname);
+
+        const favoriteAddButtons = item.element.querySelectorAll(
+          `[${LIST_ELEMENT_ATTRIBUTE}="${FAVORITE_ADD_BUTTON}"]`
+        );
+        const favoriteRemoveButtons = item.element.querySelectorAll(
+          `[${LIST_ELEMENT_ATTRIBUTE}="${FAVORITE_REMOVE_BUTTON}"]`
+        );
+
+        if (isFavorited) {
+          item.element.classList.add(FAVORITE_CLASS);
+        } else {
+          item.element.classList.remove(FAVORITE_CLASS);
+        }
+
+        for (const button of favoriteAddButtons) {
+          button.classList.toggle(FAVORITE_DISABLED_CLASS, isFavorited);
+        }
+
+        for (const button of favoriteRemoveButtons) {
+          button.classList.toggle(FAVORITE_DISABLED_CLASS, !isFavorited);
+        }
+      }
+    },
+    0
+  );
+
+  const itemsCleanup = watch(list.items, (items: ListItem[]) => handler({ items }));
+  const favoritedItemsCleanup = watch(
+    () => favoritedItemsStore[instance],
+    (favoritedItems: string[] = []) => handler({ favoritedItems }),
+    { immediate: true, deep: true }
+  );
+
+  return () => {
+    itemsCleanup();
+    favoritedItemsCleanup();
+  };
+};
+
+/**
+ * Initializes a list that displays favorited items.
+ * @param list
+ * @param instance
+ * @returns A cleanup function
+ */
+const initDisplayerFavoritesList = (list: List, instance: string) => {
+  let renderedItems = new Set<string>();
+
+  const handler = debounce(async (favoritedItems: string[] = []) => {
+    const itemsToPreserve = list.items.value.filter((item) => {
+      if (!item.url?.pathname) return false;
+
+      return favoritedItems.includes(item.url.pathname);
+    });
```

Comment:
@alexiglesias93 please double check this, I think with this new update it returns empty when toggling from existing items